git_machete/client.py

```diff
@@ -1539,7 +1539,14 @@ def __sync_annotations_to_definition_file(self,
                 if pr.user != current_user:
                     anno += f' ({pr.user})'
                 upstream: Optional[LocalBranchShortName] = self.up_branch.get(LocalBranchShortName.of(pr.head))
-                if pr.base != upstream:
+                if upstream is not None:
+                    assert upstream is not None  # Cast away 'Optional'
```

Comment:
Hmm IIRC from how mypy behaves... this shouldn't be necessary (i.e. mypy should "remember" that `upstream` is non-None after the `if`-check)

Reply:
@PawelLipski I though so too but unfortunately `assert ...` is necessary

Reply:
Wow, i don't know how but it passes now, i don't remember which Python version it was